pennylane/ops/op_math/controlled_ops.py

```diff
@@ -17,12 +17,130 @@
 
 import warnings
 from typing import Iterable
+from functools import lru_cache
 
+import numpy as np
+
+import pennylane as qml
 from pennylane.operation import AnyWires
+from pennylane.ops.qubit.non_parametric_ops import PauliY
 from pennylane.ops.qubit.matrix_ops import QubitUnitary
+from pennylane.ops.qubit.non_parametric_ops import S
 from .controlled import ControlledOp
 
 
+class CY(ControlledOp):
+    r"""CY(wires)
+    The controlled-Y operator
+
+    .. math:: CY = \begin{bmatrix}
+            1 & 0 & 0 & 0 \\
+            0 & 1 & 0 & 0\\
+            0 & 0 & 0 & -i\\
+            0 & 0 & i & 0
+        \end{bmatrix}.
+
+    .. note:: The first wire provided corresponds to the **control qubit**.
+
+    **Details:**
+
+    * Number of wires: 2
+    * Number of parameters: 0
+
+    Args:
+        wires (Sequence[int]): the wires the operation acts on
+    """
+    num_wires = 2
+    """int: Number of wires that the operator acts on."""
+
+    num_params = 0
+    """int: Number of trainable parameters that the operator depends on."""
+
+    ndim_params = ()
+    """tuple[int]: Number of dimensions per trainable parameter that the operator depends on."""
+
+    grad_method = None
+    """Gradient computation method."""
+
+    def __init__(self, wires, do_queue=True, id=None):
+        control_wire, wire = wires
+        base = PauliY(wire)
+
+        super().__init__(base, control_wire, do_queue=do_queue, id=id)
+        self._name = "CY"
+
+    def adjoint(self):
+        return CY(self.wires)
```

Comment:
That's what I thought too.

Reply:
yes it does! I'm just asking if we need to. Comparing to `simplify` right below this, we don't create a new instance, but that makes sense to me. The danger here is queuing, as it usually is with PennyLane development. Consider this:

```pycon
>>> with qml.queuing.AnnotatedQueue() as q:
...     qml.CY([0, 1]).adjoint()
```

Do you expect to see 1 instance of CY, or 2? Personally, I'd expect 1. However, there are 2. This is also true on master, but I think we should change that behaviour here. Unfortunately, `return self` doesn't queue it, so it can get messy... one possible solution is to do:
```python
qml.QueuingManager.append(self)
return self
```
Let me think on it.

Reply:
while this is of dubious behaviour, I see that it occurs all over the codebase, so the decision won't be made in this PR. Ignore me, carry on!

Reply:
Given `Adjoint.decomposition` is just `[op.adjoint()]`, and `decomposition` has the assumption that everything has to queue properly, therefore `op.adjoint()` needs to queue things too.

Reply:
`qml.simplify` handles the queuing behavior of `op.simplify`.